src/app/panels/measurements/MeasurementsTable.tsx

```diff
@@ -0,0 +1,232 @@
+/** @jsxImportSource @emotion/react */
+import { css } from '@emotion/react';
+import { useLayoutEffect, useRef } from 'react';
+
+import {
+  MeasurementBase,
+  MeasurementKind,
+  useAppDispatch,
+  useAppState,
+} from '../../../app-data/index';
+
+import MeasurementColorPreview from './MeasurementColorPreview';
+import MeasurementVisibilityToggle from './MeasurementVisibilityToggle';
+
+export interface MeasurementsTableProps {
+  kind: MeasurementKind;
+}
+
+interface MeasurementsTableRowProps {
+  item: MeasurementBase;
+  kind: MeasurementsTableProps['kind'];
+}
+
+const measurementsTableCss = {
+  root: css`
+    font-size: 0.875rem;
+    line-height: 1.25rem;
+    border-collapse: collapse;
+    table-layout: fixed;
+    width: 100%;
+  `,
+  th: css`
+    border-bottom-width: 1px;
+    font-weight: 500;
+    text-align: left;
+    padding-bottom: 10px;
+    padding-top: 10px;
+    padding-left: 2rem;
+  `,
+  tbody: css`
+    background-color: white;
+  `,
+  tr: css`
+    height: 50px;
+    max-height: 50px;
+    overflow: hidden;
+    cursor: pointer;
+    & {
+      padding-left: 2rem;
+      padding-bottom: 10px;
+      padding-top: 10px;
+      border-bottom-width: 1px;
+
+      font-weight: 500;
+      font-size: 0.875rem;
+      line-height: 1.25rem;
+    }
+  `,
+  header: css`
+    & {
+      border-bottom-width: 1px;
+      font-weight: 500;
+      text-align: left;
+      padding-bottom: 10px;
+      padding-top: 10px;
+      padding-left: 2rem;
+    }
+    & :first-child {
+      width: 70px;
+    }
+    & :nth-child(2) {
+      width: 150px;
+    }
+  `,
+  iconsContainer: css`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    justify-items: center;
+    height: 50px;
+    flex-direction: row;
+    gap: 0.5rem;
+    cursor: default;
+  `,
+  checkbox: css`
+    color: #6366f1;
+    border-color: #d1d5db;
+    border-radius: 0.25rem;
+    width: 1rem;
+    height: 1rem;
+    border-width: 1px;
+  `,
+};
+
+export function MeasurementsTable(props: MeasurementsTableProps) {
+  const { kind } = props;
+
+  const {
+    data: { measurements },
+  } = useAppState();
+
+  return (
+    <table css={measurementsTableCss.root}>
```

Comment:
We will merge this for now, but in the future we will have to use the Table component again, so the improvements need to be in the Table component, not only in MeasurementsTable.